### PR TITLE
RET-1196 Create separate userfs and systemfs filesystems

### PR DESCRIPTION
**Pull request by vegano1**
[RET-1196](https://opentrons.atlassian.net/browse/RET-1196)

Overview:
Update the opentrons-image-ot3 recipe to create two filesystems, a systemfs and a userfs
The systemfs contains the original rootfs tree without home/ var/, this will later on become read-only
The userfs contain the home/ and var/ directories and will be mounted on a separate partition

### CODE REVIEW

**Comment by DerekMaggio** [comment]
How come we want separate filesystems?

**Comment by vegano1** [comment]
> Noob question, how come we want separate filesystems?

We want to have a read-only partition (systemfs) for our own software and a read-write partition (userfs) for user data. The idea is that 
1. it will be safer to lock the parts of the system we don't want users to mess with
2. it will make it easier to do a factory reset since all the user data will be in a separate partition.

File: recipes-images/images/opentrons-ot3-image.bb [hunk 1]
```diff
@@ -53,6 +36,54 @@ IMAGE_INSTALL += " \
...
+    printf "${DISTRO_NAME} ${DISTRO_VERSION} (${DISTRO_CODENAME}) %%h\n" > ${IMAGE_ROOTFS}/etc/issue.net
+    printf "${IMAGE_NAME}\n\n" >> ${IMAGE_ROOTFS}/etc/issue
+    printf "${IMAGE_NAME}\n\n" >> ${IMAGE_ROOTFS}/etc/issue.net
+}
+# add the rootfs version to the welcome banner
+ROOTFS_POSTPROCESS_COMMAND += "add_rootfs_version;"
+
+fakeroot do_create_filesystem_trees() {
+    # this will create the systemfs tree
+    rsync -a ${IMAGE_ROOTFS}/ ${SYSTEMFS_DIR} --exclude 'home/' --exclude 'var/' --delete-excluded
```
Comment (sfoster1) [inline recipes-images/images/opentrons-ot3-image.bb:64]: love the use of rsync

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Looks good overall but we should do a pass and make sure we don't have duplications between the machine config and the image recipe

File: recipes-images/images/opentrons-ot3-image.bb [hunk 2]
```diff
@@ -47,12 +30,56 @@ IMAGE_INSTALL += " \
...
+USERFS_DIR = "${WORKDIR}/userfsfs"
+SYSTEMFS_OUTPUT = "${IMGDEPLOYDIR}/${IMAGE_NAME}.systemfs.ext4"
+USERFS_OUTPUT = "${IMGDEPLOYDIR}/${IMAGE_NAME}.userfs.ext4"
+
+do_create_filesystem_trees[depends] = "fakeroot-native:do_populate_sysroot"
+do_create_filesystem[depends] = "do_create_filesystem_trees"
+do_image_zip[depends] = "zip-native:do_populate_sysroot"
+addtask do_image_zip after do_image_complete before do_populate_lic_deploy
+
+add_rootfs_version () {
```
[the remark this answers is not in the document]
Comment (vegano1) [reply recipes-images/images/opentrons-ot3-image.bb:54]: ya will add this in a separate pr that deals with just the contents of ot3-system.zip, including the VERSION.json file, etc

**Review by sfoster1 — APPROVED** [review]
Nice, looks good to me!

So after this, we should get both filesystems in the tezi image?

**Comment by vegano1** [comment]
> Nice, looks good to me!
> 
> So after this, we should get both filesystems in the tezi image?

next pr will, this was prep work to generate the filesystems which tezi uses.

**Pull request merged**